src/sidebars/search/search-compact.less

```diff
@@ -24,6 +24,14 @@
         }
       }
     }
+
+    &.search {
+      .search-result {
+        &.selected.dark {
+          background-color: @light-background;
```

Comment:
Good point. I've pushed up a new commit that fixes something, so I don't think it applies now.